MegaQuests/src/main/java/com/elmer/megaquests/managers/QuestManager.java

```diff
@@ -13,39 +13,38 @@
 import java.io.File;
 import java.io.IOException;
 import java.math.BigDecimal;
+import java.util.EnumMap;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.UUID;
+import java.util.logging.Level;
 
 public class QuestManager {
+    private final Map<UUID, Map<Quests, Integer>> playerProgressMap = new HashMap<>();
+    private final MegaQuests megaQuests;
     private File questProgressFile;
-    Map<UUID, Map<Quests, Integer>> playerProgressMap;
-
-    MegaQuests megaQuests;
     private int questGUIAmount = 5;
 
-    public QuestManager(MegaQuests megaQuests){
+    public QuestManager(MegaQuests megaQuests) {
         this.megaQuests = megaQuests;
-        playerProgressMap = new HashMap<>();
     }
-    public void checkCompletion(Player player, Quests quests, int progress){
-        addProgress(player.getUniqueId(), quests,progress);
+
+    public void checkCompletion(Player player, Quests quests, int progress) {
+        addProgress(player.getUniqueId(), quests, progress);
 
         player.sendMessage("Checked Progress");
-        if (megaQuests.getQuestManager().getProgress(player.getUniqueId(), quests) == quests.getTaskAmount()){
+        if (megaQuests.getQuestManager().getProgress(player.getUniqueId(), quests) == quests.getTaskAmount()) {
             BigDecimal rewardAmount = new BigDecimal(quests.getReward());
 
             try {
                 megaQuests.getEssentials().getUser(player).giveMoney(rewardAmount);
-            } catch (MaxMoneyException ex) {
-                throw new RuntimeException(ex);

```

Comment:
Just the output of the `RuntimeException` doesn't tell you anything. 
`MaxMoneyException` is, you might say, standard behaviour. So it's better not to throw an error, but to inform the player that he has a money limit. Alternatively, handle the error in some other way than just throwing it in the console